subfun/BIDS_RFX.m

```diff
@@ -76,17 +76,17 @@ function BIDS_RFX(action, mmFunctionalSmoothing, mmConSmoothing, opt)
                 % FFX Directory
                 ffxDir = getFFXdir(subNumber, mmFunctionalSmoothing, opt);
                 conImg = spm_select('FPlist', ffxDir, '^con*.*nii$');
-                matlabbatch{1}.spm.spatial.smooth.data = cellstr(conImg);
+                matlabbatch{counter}.spm.spatial.smooth.data = cellstr(conImg);
+
+                % Define how much smoothing is required
+                matlabbatch{counter}.spm.spatial.smooth.fwhm = [mmConSmoothing mmConSmoothing mmConSmoothing];
+                matlabbatch{counter}.spm.spatial.smooth.dtype = 0;
+                matlabbatch{counter}.spm.spatial.smooth.prefix = [...
+                spm_get_defaults('smooth.prefix'), num2str(mmConSmoothing)];
```

Comment:
again , same problem with matlabbatch{1}. 
matlabbatch{1} --> was having the settings that are specified in the script
matlabbatch{2:end} --> was having the default setting which didnt match. 

the matlabbatch{counter} will fix this, and all subjects will have the same parameters and settings.